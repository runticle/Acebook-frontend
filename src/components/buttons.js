import React from 'react';


class Buttons extends React.Component {

  deletePostReq() {
    fetch("https://acebook-stars.herokuapp.com/posts", {
      method: 'delete',
      id: this.props.post_id
    })
      .then(res => res.json())
      .then(
        (result) => {
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          this.setState({
            error
          });
        }
      )
  }


  handleDelete = event => {
    event.preventDefault();
    this.deletePostReq();
  }

  handleEdit = event => {
    event.preventDefault();
    console.log(this.props.post_id)
  }

  handleComment = event => {
    event.preventDefault();
    console.log(this.props.post_id)
  }

  handleLike = event => {
    event.preventDefault();
    console.log(this.props.post_id)
  }

  handleMore = event => {
    event.preventDefault();
    console.log(this.props.post_id)
  }

  render() {

    return (
              <div id="buttons">
                <button id="delete" onClick={ this.handleDelete } className="button">❌</button>
                <button id="edit" onClick={ this.handleEdit } className="button">✏️</button>
                <button id="new_comment_button" onClick={ this.handleComment } className="button">💬</button>
                <button id="toggle_like" onClick={ this.handleLike } className="button">⭐️</button>
                <button id="see_more" onClick={ this.handleMore } className="button">➕</button>
              </div>
          )
    }
}

export default Buttons;
