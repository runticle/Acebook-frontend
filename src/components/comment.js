import React from 'react';
import AllCommentButtons from './comment_button/allCommentButtons';
import NewComment from './newComment'
import Time from 'react-time-format';


export class Comment extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    return (
      <div name="comment" id="comment_id">
        { this.renderNameTime() }
        { this.renderMessage() }
        < AllCommentButtons
          post_id={ this.props.post_id }
          id={this.props.id}
        />
      </div>
      )
    }

    renderNameTime() {
      return (
      <div id="name_time">
        <h4 className="user_name">User #{this.props.user}</h4>
        <p id="time"> <Time value={ this.props.time } format="hh:mm YYYY/MM/DD" /> </p>
      </div>
    )
    }

    renderMessage() {
      return (
      <div id="message">
        <p>{this.props.message}</p>
      </div>
    )
    }
}

export default Comment;
