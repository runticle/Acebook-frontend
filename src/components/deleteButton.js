import React from 'react';


class DeleteButton extends React.Component {

  // deletePostReq() {
  //   fetch("https://acebook-stars.herokuapp.com/posts", {
  //     method: 'delete',
  //     id: this.props.post_id
  //   })
  //     .then(res => res.json())
  //     .then(
  //       (result) => {
  //         console.log(result)
  //       },
  //       // Note: it's important to handle errors here
  //       // instead of a catch() block so that we don't swallow
  //       // exceptions from actual bugs in components.
  //       (error) => {
  //         console.log(error)
  //         this.setState({
  //           error
  //         });
  //       }
  //     )
  // }

  handleDelete = event => {
    event.preventDefault();
    console.log(this.props.post_id)
    // this.deletePostReq();
  }

  render() {

    return (
      <div id="deleteButton">
        <button id="delete" onClick={ this.handleDelete } className="button">‚ùå</button>
      </div>
          )
    }
}

export default DeleteButton;
