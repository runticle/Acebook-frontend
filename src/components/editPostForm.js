import React from 'react';

export class EditPostForm extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      updated_message: ""
    };
  }

  editPostReq() {
    fetch(`https://acebook-stars.herokuapp.com/posts/${this.props.post_id}`, {
      method: 'patch',
      message: this.state.updated_message,
      // user_id: // USER ID FROM TOKEN!!!!! WOOO!
    })
      .then(res => res.json())
      .then(
        (result) => {
          console.log('result!')
          console.log(result)
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          this.setState({
            error
          });
        }
      )
  }

  validateForm() {
    return this.state.updated_message.length > 0
  }

  handleChange = event => {
    this.setState({
      [event.target.className]: event.target.value
    });
  }

  handleSubmit = event => {
    event.preventDefault();
    this.editPostReq();
  }

  render() {
    return (
      <div className="new_comment">
        <form className="edit_message_form" onSubmit={this.handleSubmit}>
          <textarea className="updated_message" onChange={this.handleChange} type = "text" value= {this.props.message}>  </textarea>
          <button className="submit_message" disabled={!this.validateForm()}>Submit</button>
        </form>
      </div>
    )
  }
}

export default EditPostForm;
