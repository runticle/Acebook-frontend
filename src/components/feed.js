import React from 'react';
import ReactDOM from 'react-dom';
import Post from './post';

class Feed extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      isLoaded: false,
      posts: []
    };
  }

  componentDidMount() {
    fetch("https://acebook-stars.herokuapp.com/posts")
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            isLoaded: true,
            posts: result
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      )
  }

  render() {
    const { error, isLoaded, posts } = this.state;
      if (error) {
        return <div>Error: {error.message}</div>;
      } else if (!isLoaded) {
        return <div>Loading...</div>;
      } else {
        return (
          <ul>
          { this.state.posts.map((post, i) => (
            < Post
              key={i}
              message={post.message}
              time={post.created_at}
              user={post.user_id}
            />
          ))}
          </ul>
        )
      }
  }

}

export default Feed;
