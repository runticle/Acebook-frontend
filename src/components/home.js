import React from 'react';
import Login from './login';
import Register from './register';
import Welcome from './welcome';
import Navigation from './navigation';
import {Redirect} from 'react-router'


class Home extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      loggedIn: false,
      loginHidden: false,
      registerHidden: true
    };
  }

  logoutReq() {
    fetch("https://acebook-stars.herokuapp.com/users/sign_out", {
      method: 'delete',
      // user_id: // USER ID FROM TOKEN!!!!! WOOO!
    })
      .then(res => res.json())
      .then(
        (result) => {
          console.log(result)
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          console.log(error)
          this.setState({
            error
          });
        }
      )
  }


    handleRegister = event => {
      event.preventDefault();
      document.getElementById('register_link').style.display = 'none'
      this.setState({loginHidden:true})
      this.setState({registerHidden:false})
    }

  render() {
    document.body.style = "background-color: #e6e9ef;"

    return (
      <div>
        <Navigation />
        < Welcome />
        <div id="Authentication">
        <div id="login_box" className="form">
          <Login
            loginHidden={this.state.loginHidden}
          />
          <p id="register_link" className="form"> New user? <a href='/' onClick={this.handleRegister}>Register</a></p>
        </div>
        <div id="register_box" className="form">
          <Register
            registerHidden={this.state.registerHidden}
          />
        </div>
        </div>
      </div>
    )
  }

}

export default Home;
