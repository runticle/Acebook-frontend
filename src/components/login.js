import React from 'react';

class Login extends React.Component {
  render () {
    return (
      <div id="login">
        <input type="email" placeholder="Email"/>
        <input type="password" placeholder="Password"/>
        <button id="submit_login">Submit</button>
      </div>
    )
  }
}

export default Login;
