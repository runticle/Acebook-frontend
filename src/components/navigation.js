import React from 'react';
import Login from './login';
import Register from './register';


class Navigation extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      logged_in: false,
      loginHidden: false,
      registerHidden: true
    };
  }

  handleRegister = event => {
    event.preventDefault();
    document.getElementById('register_link').style.display = 'none'
    this.setState({loginHidden:true})
    this.setState({registerHidden:false})
  }

  render() {
    return (
      <div>
        <nav className="navbar navbar-dark" id="navbar">
          <h2>Acebook</h2>
          <a href='/'>Timeline</a>
          <a href='/'>Logout</a>
        </nav>
        <div id="Authentication">
        <div id="login_box" className="form">
          <Login
            loginHidden={this.state.loginHidden}
          />
          <p id="register_link" className="form"> New user? <a href='/' onClick={this.handleRegister.bind(this)}>Register</a></p>
        </div>
        <div id="register_box" className="form">
          <Register
            registerHidden={this.state.registerHidden}
          />
        </div>
        </div>
      </div>
    )
  }

}

export default Navigation;
