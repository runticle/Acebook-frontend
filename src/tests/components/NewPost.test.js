import React from 'react'
import { shallow } from 'enzyme'
import { NewPost } from '../../components/NewPost'

let wrapper = shallow(<NewPost />)

test('should render NewPost correctly', () => {
  expect(wrapper).toMatchSnapshot();
})
