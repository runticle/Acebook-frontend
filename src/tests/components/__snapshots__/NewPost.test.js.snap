// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NewPost correctly 1`] = `
<form
  className="new_message_form"
  onSubmit={[Function]}
>
  <textarea
    id="post_message"
    placeholder="What's on your mind...?"
  />
  <button
    id="submit_message"
  >
    Submit
  </button>
</form>
`;
